src/rust/engine/process_execution/src/lib.rs

```diff
@@ -269,6 +273,29 @@ pub trait CommandRunner: Send + Sync {
   }
 }
 
+// TODO(#8513) possibly move to the MEPR struct, or to the hashing crate?
+pub fn digest(
+  req: MultiPlatformExecuteProcessRequest,
+  metadata: &ExecuteProcessRequestMetadata,
+) -> Digest {
+  let mut hashes: Vec<String> = req
+    .0
+    .values()
+    .map(|ref epr| crate::remote::make_execute_request(epr, metadata.clone()).unwrap())
```

Comment:
This doesn't fingerprint the JDK path or version, which it probably should for your nailgun purposes, but definitely shouldn't for remote execution purposes.

Reply:
Solved here: https://github.com/pantsbuild/pants/pull/8371/commits/288040b800d422a0015deb5960f55895d21dd08b